CIScripts/Tests/TCPCommunicationTest.ps1

```diff
@@ -1,3 +1,67 @@
 function Test-TCPCommunication {
-    # TODO: JW-838: Implement
+    Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session,
+           [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
+
+    Write-Host "Running TCP Communication test"
+
+    Initialize-TestConfiguration -Session $Session -TestConfiguration $TestConfiguration
+
+    $NetworkName = $TestConfiguration.DockerDriverConfiguration.NetworkConfiguration.NetworkName
+
+    Write-Host "Creating containers"
+    $ServerID, $ClientID = Invoke-Command -Session $Session -ScriptBlock {
+        $ServerID = docker run --network $Using:NetworkName -d iis-tcptest
+        $ClientID = docker run --network $Using:NetworkName -d microsoft/nanoserver ping -t localhost
+        return $ServerID, $ClientID
+    }
+
+    . $PSScriptRoot\CommonTestCode.ps1
+
+    Write-Host "Getting VM NetAdapter Information"
+    $VMNetInfo = Get-RemoteNetAdapterInformation -Session $Session -AdapterName $TestConfiguration.AdapterName
+
+    Write-Host "Getting Containers NetAdapter Information"
+    $ServerNetInfo = Get-RemoteContainerNetAdapterInformation -Session $Session -ContainerID $ServerID
+    $ClientNetInfo = Get-RemoteContainerNetAdapterInformation -Session $Session -ContainerID $ClientID
+
+    Write-Host "Configuring..."
```

Comment:
A comment, which route you are configuring would be helpfull here. Like
```
Write-Host "Setting a connection between $Using:ServerNetInfo.MACAddress and $Using:ClientNetInfo.MACAddress
```
This would, I believe, simplify debugging when something goes wrong.